enterprise/cluster/src/main/java/org/neo4j/cluster/com/NetworkSender.java

```diff
@@ -424,6 +426,8 @@ public ChannelPipeline getPipeline() throws Exception
     private class NetworkMessageSender
             extends SimpleChannelHandler
     {
+        private Throwable lastException;
```

Comment:
This methods looked a lot like callbacks to me that is why I though so. It is however not true:
https://github.com/netty/netty/blob/netty-3.6.3.Final/src/main/java/org/jboss/netty/channel/ChannelUpstreamHandler.java#L70
and both `exceptionCaught` and `writeComplete` are called from `ChannelUpstreamHandler#handleUpstream`.